lib/ansible/modules/network/meraki/meraki_rest.py

```diff
@@ -0,0 +1,238 @@
+        mutually_exclusive=[['content', 'src']],
+    )
+    module.params['follow_redirects'] = 'urllib2'
+
+    try:
+        module.params['auth_key'] = os.environ['MERAKI_KEY']
+    except KeyError:
+        pass
+
+    if module.params['auth_key'] is None:
+        module.fail_json(msg='Meraki Dashboard API key not set')
+
+    path = module.params['path']
+    payload = None
+
+    if module.params['content']:
+        payload = module.params['content']
+        payload = json.dumps(payload)
+    elif module.params['src']:
+        src = module.params['src']
+        file_exists = False
+        if os.path.isfile(src):
+            file_exists = True
+            with open(src, 'r') as config_object:
+                payload = config_object.read()
+        else:
+            module.fail_json(msg="File does not exist")
+
+    # if the user is working with this module in only check mode we do not
+    # want to make any changes to the environment, just return the current
+    # state with no modifications
+    if module.check_mode:
+        return result
+
+    # manipulate or modify the state as needed (this is going to be the
+    # part where your module will do what it needs to do)
+    protocol = 'https'
+    if module.params['use_ssl'] is not None and module.params['use_ssl'] is False:
+        protocol = 'http'
+
+    url = '{0}://{1}/{2}'.format(protocol, module.params['host'], module.params['path'].lstrip('/'))
+    headers = {'Content-Type': 'application/json',
+               'X-Cisco-Meraki-API-Key': module.params['auth_key'],
+               }
+
+    if module.params['output_level'] == 'debug':
+        result['url'] = url
+        result['method'] = module.params['method']
+        result['headers'] = headers
+        result['payload'] = payload
+
+    try:
+        resp, info = fetch_url(module, url,
+                               data=payload,
+                               headers=headers,
+                               method=module.params['method'].upper(),
+                               use_proxy=module.params['use_proxy'],
+                               force=False,
+                               timeout=module.params['timeout'],
+                               )
+        if module.params['output_level'] == 'debug':
+            result['response_status_code'] = str(info['status'])
+            result['response_headers'] = str(resp.headers)
+
+    except Exception as e:
+        module.fail_json(msg=e)
+
+    if info['status'] >= 300:
+        module.fail_json(msg='{0}: {1} '.format(info['status'], info['body']), **result)
+
+    # use whatever logic you need to determine whether or not this module
+    # made any modifications to your target
+    if info['status'] >= 200 and info['status'] <= 299:
+        result['changed'] = True
+        try:
+            result['message'] = json.loads(to_native(resp.read()))
+        except:
+            module.fail_json(msg="Meraki dashboard didn't return JSON compatible data")
```

Comment:
It's not Meraki dashboard, but Meraki API that didn't return JSON compatible data.

We have to get rid of any relation to Dashboard IMO. Dashboard is the GUI.

Reply:
Once again, I'm conflicted on the nomenclature I should use. Meraki has multiple APIs and this one in particular is called Dashboard API. I need to think it through a little more but Meraki Dashboard API may be the best bet. Otherwise it's ambiguous to which of Meraki's APIs it may refer to. I'll make a change though before it's merged though.